metrics-core/src/main/java/com/codahale/metrics/ChunkedAssociativeLongArray.java

```diff
@@ -1,95 +1,87 @@
 package com.codahale.metrics;
 
+import static java.lang.System.arraycopy;
 import static java.util.Arrays.binarySearch;
 
-import java.util.ArrayDeque;
-import java.util.Deque;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.ListIterator;
 
 class ChunkedAssociativeLongArray {
     private static final long[] EMPTY = new long[0];
     private static final int DEFAULT_CHUNK_SIZE = 512;
 
     private final int defaultChunkSize;
-    private Chunk activeChunk;
+    private final LinkedList<Chunk> chunks = new LinkedList<Chunk>();
```

Comment:
LinkedList here has relatively small size 
`countOfStoredMeasurements / DEFAULT_CHUNK_SIZE`.
And we are heavily really on LinkedList implementation because:
1. Now we deleting chunks from both sides of the list  in` trim(long startKey, long endKey)` 
2. Deleting from and inserting chunks into the middle in `clear(long startKey, long endKey)`
And `LinkedList` give us O(1) complexity for all this operations.